File: go/gcdOfStrings_test.go
// https://leetcode.com/problems/greatest-common-divisor-of-strings/

package leetcode

import "testing"

func Test_gcdOfStrings(t *testing.T) {
	type args struct {
		str1 string
		str2 string
	}
	tests := []struct {
		name string
		args args
		want string
	}{
		{
			name: "Example 1",
			args: args{"ABCABC", "ABC"},
			want: "ABC",
		},
		{
			name: "Example 2",
			args: args{"ABABAB", "ABAB"},
			want: "AB",
		},
		{
			name: "Example 3",
			args: args{"LEET", "CODE"},
			want: "",
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := gcdOfStrings(tt.args.str1, tt.args.str2); got != tt.want {
				t.Errorf("gcdOfStrings() = %v, want %v", got, tt.want)
			}
		})
	}
}

File: python/src/gcdOfStrings_test.py
import pytest

from gcdOfStrings import Solution

solution = Solution()


@pytest.mark.parametrize(
    "fn",
    (
        (solution.gcdOfStrings),
        (solution.gcdOfStrings_python),
    ),
)
@pytest.mark.parametrize(
    "str1, str2, want",
    (
        ("ABCABC", "ABC", "ABC"),
        ("ABABAB", "ABAB", "AB"),
        ("LEET", "CODE", ""),
    ),
)
def test_gcdOfStrings(fn: callable, str1: str, str2: str, want: str):
    got = fn(str1, str2)
    assert got == want
